scripts/lib/authority-and-program-updater.ts

```diff
@@ -157,11 +211,21 @@ export class AuthorityAndProgramUpdater {
           name: authority.name,
         };
     }
-    fs.writeFileSync(
-      AUTHORITIES_JSON_FILE,
-      JSON.stringify(json, null, 2),
-      'utf-8',
-    );
+
+    // Reset file first.
+    fs.writeFileSync(AUTHORITIES_JSON_FILE, '', 'utf-8');
+    // Go through list of state abbreviations and then write them, if entries exist.
+    for (const entry_state of stateAbbreviationArray) {
```

Comment:
So I assume this has to do with the weirdness that I didn't understand when we were chatting, but are we sure something like https://stackoverflow.com/questions/5467129/sort-javascript-object-by-key doesn't work?

It seems like we should be able to get the keys of the json, sort them, then iterate over them populating a new object with the key->value pairs. And then write overwrite the original file with that object.

If that doesn't work for some reason and we have to iterate over the abbreviation list, I'd still recommend collecting the stringified states into a single object and then doing a single write. That will ease the testing story (I'll respond to your big-picture testing comment elsewhere).

Reply:
Realized https://stackoverflow.com/questions/5467129/sort-javascript-object-by-key will work if I `type` everything; so may not work for programs as-is, but assuming we can either create a separate sort for program's JSON or modify type to take either. Thanks! Will add tests now.